problem_builder/mentoring.py

```diff
@@ -776,3 +789,98 @@ def workbench_scenarios():
         Scenarios displayed by the workbench. Load them from external (private) repository
         """
         return loader.load_scenarios_from_path('templates/xml')
+
+
+class MentoringWithExplicitStepsBlock(BaseMentoringBlock):
+    """
+    An XBlock providing mentoring capabilities with explicit steps
+    """
+    # Settings
+    display_name = String(
+        display_name=_("Title (Display name)"),
+        help=_("Title to display"),
+        default=_("Mentoring Questions (with explicit steps)"),
+        scope=Scope.settings
+    )
+
+    editable_fields = ('display_name',)
+
+    @lazy
+    def questions(self):
+        """ Get the usage_ids of all of this XBlock's children that are "Questions" """
+        return list(chain.from_iterable(self.runtime.get_block(step_id).steps for step_id in self.steps))
+
+    @property
+    def steps(self):
+        """
+        Get the usage_ids of all of this XBlock's children that are "Steps"
+        """
+        from .step import MentoringStepBlock  # Import here to avoid circular dependency
+        return [
+            _normalize_id(child_id) for child_id in self.children if
+            child_isinstance(self, child_id, MentoringStepBlock)
+        ]
+
+    def student_view(self, context):
+        fragment = Fragment()
+        child_content = u""
+
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if child is None:  # child should not be None but it can happen due to bugs or permission issues
+                child_content += u"<p>[{}]</p>".format(self._(u"Error: Unable to load child component."))
+            elif not isinstance(child, MentoringMessageBlock):
+                try:
```

Comment:
@itsjeyd this try-catch block can be replaced by [`StudioContainerWithNestedXBlocksMixin._render_child_fragment`](https://github.com/edx/xblock-utils/blob/8f4e7689367342f3637712fa871f870c66fed19b/xblockutils/studio_editable.py#L476):

    child_fragment = self._render_child_fragment(child, context, view='mentoring_view')